Source/ProgressionSystemRuntime/Public/Data/PSDataAsset.h

```diff
@@ -116,19 +116,18 @@ class PROGRESSIONSYSTEMRUNTIME_API UPSDataAsset : public UMyPrimaryDataAsset
 	UPROPERTY(EditAnywhere, Category = "UI|Material", meta = (BlueprintProtected, DisplayName = "Dynamic Progression Material"))
 	TObjectPtr<class UMaterialInterface> DynamicProgressionMaterialInternal = nullptr;
 
+	/** Material applied as  dynamic progression overlay material (star is filled partially depends on progression) */
+	UPROPERTY(EditAnywhere, Category = "UI|Material", meta = (BlueprintProtected, DisplayName = "Dynamic Progression Overlay Material"))
+	TObjectPtr<class UMaterialInterface> DynamicProgressionOverlayMaterialInternal = nullptr;
```

Comment:
In this review, I see you created next new properties:
- Material Overlay in Data Asset
- transient `StarOverlayProgressionMaterial` and `StarMaterialOverlayMap` in World Subsystem

But the idea is refactor already existing Progression Material Instances to apply as Overlay Material instead of regular material apply.

Please dont create new properties, but make your existing material properties (such as `StarDynamicProgressMaterial`, `StarLockedProgressMaterial`, `StarUnLockedProgressMaterial`) to apply as Overlay.

As result, your logic almost stay the same as before review, but only `SetMaterial` will change with `SetOverlayMaterial`

Reply:
Before I've reached this comment I've deleted the mentioned materials and instance. So now we have only 1 set of material and material instance  `DynamicProgressionBombOverlayMaterialInternal'. 
Overlay material needs to be develop and replaced by designer. Attached reference image below. 
![image](https://github.com/user-attachments/assets/f8b71ff7-523b-4fb5-99cc-9440d2126ebc)